### PR TITLE
[mongodb] fix connections_current metric mongodb-Replica-Set

### LINKED ISSUE #466: mackerel-agent-plugins-mongodb Replica-Set was wrong value
Hi, I'm new in mackerel and golang.
Recently I apply mackerel-agent-plugins-mongodb on my database-system using Replica-Set:
https://docs.mongodb.com/manual/core/replica-set-members/

* OS: CentOS 6.3
* DB: mongodb 3.2
   * db001 - PRIMARY
   * db002 - SECONDARY
   * db003 - SECONDARY

The `connections_current` metric was wrong value:
* Mackerel-monitor chart:
    * db001: connections_current: 40
    * db002: connections_current: 40
* Linux shell command

```
[root@db001:~]# mongo
MongoDB shell version: 3.2.10
connecting to: test
rs1:PRIMARY> db.serverStatus()
{
	"connections" : {
		"current" : 40,
		"available" : 51160,
		"totalCreated" : NumberLong(12263755)
	}
}
```

```
[root@db002:~]# mongo
MongoDB shell version: 3.2.10
connecting to: test
rs1:SECONDARY> db.serverStatus()
{
        "connections" : {
		"current" : 60,
		"available" : 51140,
		"totalCreated" : NumberLong(1773812)
	}
}
```
The connections_current metric at SECONDARY-server alway follow PRIMARY-server. That was wrong value.

I try to using Monotonic mode, with connect=direct:
```
session.SetMode(mgo.Monotonic, true)
mongodb.URL = fmt.Sprintf("mongodb://%s:%s/?connect=direct", *optHost, *optPort)
```
And it work ok.

### PR DESCRIPTION
**Pull request by vfa-cancc**
fix #466

### CODE REVIEW

**Comment by astj** [comment]
Confirmed with following docker-compose.yml: https://gist.github.com/astj/a4591bab5eec8d1af8aafa3754462f2d

**Comment by astj** [comment]
@vfa-cancc Sorry for late response. Reviewed!

**Comment by vfa-cancc** [comment]
@astj Thanks for your review and suggestions.
I have been used `mgo.DialWithInfo` and pass a `*mgo.DialInfo`
`Eventual` mode was work correctly!
Could you please help me to check this p-r again?
Thank you!

File: mackerel-plugin-mongodb/lib/mongodb.go [hunk 1]
```diff
@@ -183,12 +184,18 @@ type MongoDBPlugin struct {
 }
 
 func (m MongoDBPlugin) fetchStatus() (bson.M, error) {
-	session, err := mgo.Dial(m.URL)
+	mongoDBDialInfo := &mgo.DialInfo{
+		Addrs:    []string{m.URL},
+		Direct: true,
+		Timeout:  10 * time.Second,
```
Comment (astj) [inline mackerel-plugin-mongodb/lib/mongodb.go]: 👍

File: mackerel-plugin-mongodb/lib/mongodb.go [hunk 2]
```diff
@@ -183,12 +184,18 @@ type MongoDBPlugin struct {
 }
 
 func (m MongoDBPlugin) fetchStatus() (bson.M, error) {
-	session, err := mgo.Dial(m.URL)
+	mongoDBDialInfo := &mgo.DialInfo{
+		Addrs:    []string{m.URL},
+		Direct: true,
```
Comment (astj) [inline mackerel-plugin-mongodb/lib/mongodb.go]: Please apply `gofmt`. (We should check it in ci...)
```diff
diff --git a/mackerel-plugin-mongodb/lib/mongodb.go b/mackerel-plugin-mongodb/lib/mongodb.go
index bcc31f0..d63459e 100644
--- a/mackerel-plugin-mongodb/lib/mongodb.go
+++ b/mackerel-plugin-mongodb/lib/mongodb.go
@@ -185,9 +185,9 @@ type MongoDBPlugin struct {

 func (m MongoDBPlugin) fetchStatus() (bson.M, error) {
        mongoDBDialInfo := &mgo.DialInfo{
-               Addrs:    []string{m.URL},
-               Direct: true,
-               Timeout:  10 * time.Second,
+               Addrs:   []string{m.URL},
+               Direct:  true,
+               Timeout: 10 * time.Second,
        }
        session, err := mgo.DialWithInfo(mongoDBDialInfo)
        if err != nil {
```

File: mackerel-plugin-mongodb/lib/mongodb.go [hunk 3]
```diff
@@ -183,12 +184,18 @@ type MongoDBPlugin struct {
 }
 
 func (m MongoDBPlugin) fetchStatus() (bson.M, error) {
-	session, err := mgo.Dial(m.URL)
+	mongoDBDialInfo := &mgo.DialInfo{
+		Addrs:    []string{m.URL},
```
Comment (astj) [inline mackerel-plugin-mongodb/lib/mongodb.go]: I could not establish an authenticated connection (with `-username` and `-password`) .
```
$ go run main.go -username bob -password 1234
2017/12/27 18:19:16 OutputValues:  no reachable servers
exit status 1
```

Can you confirm it?
(If the change will be big, we can go back to the first commit https://github.com/mackerelio/mackerel-agent-plugins/pull/467/commits/be0b799079122699335a89bb26dea2a818122926 and simply change the mode to `Eventual`. )

**Comment by vfa-cancc** [comment]
@astj I have been updated with parameter options (-username and -password).
Could you please confirm it?

File: mackerel-plugin-mongodb/lib/mongodb.go (hunk 2 above)
Comment (vfa-cancc) [reply mackerel-plugin-mongodb/lib/mongodb.go]: @astj 
I have been apply gofmt.

File: mackerel-plugin-mongodb/lib/mongodb.go (hunk 3 above)
Comment (vfa-cancc) [reply mackerel-plugin-mongodb/lib/mongodb.go]: @astj 
I have been updated with parameter options (-username and -password).
Could you please confirm it?
Thanks you!

Comment (astj) [reply mackerel-plugin-mongodb/lib/mongodb.go]: Confirmed working. :+1:

**Comment by astj** [comment]
LGTM. Thank you!

**Pull request merged**